dialyzer.ignore-warnings

```diff
@@ -3,4 +3,4 @@ lib/websockex.ex:19: Function handle_cast/2 only terminates with explicit except
 lib/websockex.ex:297: Expression produces a value of type 'ok' | {'error',_}, but this value is unmatched
 
 # This is due to 3rd-party Plug.Conn.merge_resp_headers/2 potentially raising `AlreadySentError` but was not spec'ed.
-apps/potterhat_rpc/lib/potterhat_rpc/router.ex:44: The created fun has no local return
+apps/potterhat_rpc/lib/potterhat_rpc/router.ex:48: The created fun has no local return
```

Comment:
hmmm.... this might be pointing to a dialyzer error in `forward/3`!

Reply:
I think I did try removing all code from `forward/3` and replaced with `{:ok, %PotterhatNode.Node.RPCResponse{}}` and `{:error, :no_nodes_available}` and the error was still there. But let me double check...

Reply:
This is where the error came from:

```elixir
  post "/" do
    case EthForwarder.forward(conn.body_params, conn.req_headers) do
      {:ok, response} ->
        conn
        |> Map.put(:resp_headers, [])
        |> merge_resp_headers(response.headers)
        |> send_resp(response.status_code, response.body)

      {:error, code} ->
        ErrorHandler.send_resp(conn, code)
    end
  end
```

Editing to this causes dialyzer to pass:

```elixir
  post "/" do
    case EthForwarder.forward(conn.body_params, conn.req_headers) do
      {:ok, response} ->
        conn

      {:error, code} ->
        ErrorHandler.send_resp(conn, code)
    end
  end
```

but editing to this also causes dialyzer to pass:

```elixir
  post "/" do
    case EthForwarder.forward(conn.body_params, conn.req_headers) do
      {:ok, response} ->
        conn
        |> Map.put(:resp_headers, [])
        |> merge_resp_headers(response.headers)
        |> send_resp(response.status_code, response.body)

      {:error, code} ->
        conn
    end
  end
```

So I couldn't locate the problem so far. Emptying out `EthForwarder.forward/2` and `ErrorHandler.send_resp/2` also doesn't remove the dialyzer warning.

Reply:
Change 
```
@spec forward(map(), map(), Keyword.t()) ::
          {:ok, %PotterhatNode.Node.RPCResponse{}} | {:error, :no_nodes_available}
  def forward(body_params, header_params, opts \\ []) do
    node_registry = Keyword.get(opts, :node_registry, ActiveNodes)
    nodes = ActiveNodes.all(node_registry)

    do_forward(nodes, body_params, header_params)
  end
```
to 
```
@spec forward(map(), Keyword.t(), Keyword.t()) ::
          {:ok, %PotterhatNode.Node.RPCResponse{}} | {:error, :no_nodes_available}
  def forward(body_params, header_params, opts \\ []) do
    node_registry = Keyword.get(opts, :node_registry, ActiveNodes)
    nodes = ActiveNodes.all(node_registry)

    do_forward(nodes, body_params, header_params)
  end
```

`req_headers` is a list of tuples not a map.